gobblin-data-management/src/main/java/gobblin/data/management/copy/predicates/RootDirectoryModtimeSkipPredicate.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.copy.predicates;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.ql.metadata.Partition;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Predicate;
+
+import gobblin.data.management.copy.CopyContext;
+import gobblin.data.management.copy.hive.HiveCopyEntityHelper;
+import gobblin.util.PathUtils;
+
+import javax.annotation.Nullable;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+
+/**
+ * Use with {@link HiveCopyEntityHelper#FAST_PARTITION_SKIP_PREDICATE}.
+ * Skips partitions whose data location exists in the target, and such that the target location has a newer mod time
+ * than the source location.
+ */
+@AllArgsConstructor
+@Slf4j
+public class RootDirectoryModtimeSkipPredicate implements Predicate<Partition> {
+
+  private HiveCopyEntityHelper helper;
+
+  @Override
+  public boolean apply(@Nullable Partition input) {
+
+    if (input == null) {
+      return true;
+    }
+
+    try {
+
+      if (PathUtils.isGlob(input.getDataLocation())) {
```

Comment:
not now, but I don't know if they will change that in the future in Hive. This way we protect ourselves from such a change.